cloudSupport/src/main/scala/cromwell/cloudSupport/gcp/gcs/GcsStorage.scala

```diff
@@ -0,0 +1,56 @@
+package cromwell.cloudSupport.gcp.gcs
+
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport
+import com.google.api.client.googleapis.media.MediaHttpUploader
+import com.google.api.client.json.jackson2.JacksonFactory
+import com.google.api.gax.retrying.RetrySettings
+import com.google.cloud.storage.StorageOptions
+import com.google.api.services.storage.Storage
+import com.google.auth.Credentials
+import com.google.cloud.storage.contrib.nio.CloudStorageConfiguration
+import com.typesafe.config.ConfigFactory
+import cromwell.cloudSupport.gcp.GoogleConfiguration
+import cromwell.cloudSupport.gcp.http.GoogleHttpTransportOptions.TransportOptions
+import net.ceedubs.ficus.Ficus._
+
+object GcsStorage {
+  val JsonFactory = JacksonFactory.getDefaultInstance
+  val HttpTransport = GoogleNetHttpTransport.newTrustedTransport
+
+  val DefaultCloudStorageConfiguration = {
+    val UploadBufferBytes = ConfigFactory.load().as[Option[Int]]("google.upload-buffer-bytes").getOrElse(MediaHttpUploader.MINIMUM_CHUNK_SIZE)
```

Comment:
ToL consider making this a class and taking `JsonFactory`, `HttpTransport`, and `Int`  as constructor params?

Reply:
Why a singleton class rather than an object?

Reply:
@mcovarr was that to me or to @danbills ?

Reply:
@danbills

Reply:
object doesn't take parameters.  You could stick to object if every function returned a `SomeConfigType => X`, i.e. taking the dependencies as function inputs.  I.e. dependency injection via `Reader` monad.

Reply:
Watching the convo unfold, since this was just a move from a different package I'm going to raise "activation energy" as a potential blocker for suggestions here :)

Reply:
BTW that's not to say that I won't change anything, but come to a consensus easier than wiring `Reader` around :)